app/components/progress-bar.js

```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { debounce } from 'lodash';
+export default class ProgressBarComponent extends Component {
+  @tracked isEditable = false;
+  @tracked value = this.args.value;
+  lastEditTime = null;
+
+  @action turnEditModeOn() {
+    this.isEditable = true;
+    this.lastEditTime = Date.now();
+    this.setEditableToFalse();
+  }
+
+  setEditableToFalse() {
+    setTimeout(() => {
+      if (this.isEditable && Date.now() - this.lastEditTime >= 5000) {
```

Comment:
- Instead of having all the logic inside the if statement, moving the date delta into its own variable would look much nicer. 
```js
timeDelta = Date.now() - this.lastEditedTime
if (this.isEditable && timeDelta >= 5000){
    ...
}
```
what do you think?

Reply:
made required changes.